store/proof.go

```diff
@@ -96,3 +121,69 @@ func (op CommitmentOp) Run(args [][]byte) ([][]byte, error) {
 
 	return [][]byte{root}, nil
 }
+
+// ProofFromByteSlices computes the proof from the given leaves.
+func ProofFromByteSlices(leaves [][]byte, index int) (rootHash []byte, inners []*ics23.InnerOp) {
+	if len(leaves) == 0 {
+		return emptyHash(), nil
+	}
+
+	n := len(leaves)
+	for n > 1 {
+		if index < n-1 || index&1 == 1 {
+			inner := &ics23.InnerOp{Hash: ics23.HashOp_SHA256}
+			if index&1 == 0 {
+				inner.Prefix = innerPrefix
+				inner.Suffix = leaves[index^1]
+			} else {
+				inner.Prefix = append(innerPrefix, leaves[index^1]...)
+			}
+			inners = append(inners, inner)
+		}
+		for i := 0; i < n/2; i++ {
+			leaves[i] = InnerHash(leaves[2*i], leaves[2*i+1])
+		}
+		if n&1 == 1 {
+			leaves[n/2] = leaves[n-1]
+		}
+		n = (n + 1) / 2
+		index /= 2
+	}
+
+	rootHash = leaves[0]
+	return
+}
+
+// ConvertCommitmentOp converts the given merkle proof into an CommitmentOp.
+func ConvertCommitmentOp(inners []*ics23.InnerOp, key, value []byte) CommitmentOp {
+	return NewSimpleMerkleCommitmentOp(key, &ics23.CommitmentProof{
+		Proof: &ics23.CommitmentProof_Exist{
+			Exist: &ics23.ExistenceProof{
+				Key:   key,
+				Value: value,
+				Leaf:  SimpleMerkleSpec.LeafSpec,
+				Path:  inners,
+			},
+		},
+	})
+}
+
+func emptyHash() []byte {
+	h := sha256.Sum256([]byte{})
+	return h[:]
+}
+
+// LeafHash computes the hash of a leaf node.
+func LeafHash(key, value []byte) ([]byte, error) {
+	return SimpleMerkleSpec.LeafSpec.Apply(key, value)
+}
+
+// InnerHash computes the hash of an inner node.
+func InnerHash(left, right []byte) []byte {
+	data := innerPrefix
+	data = append(data, left...)
+	data = append(data, right...)
+	h := sha256.Sum256(data)
```

Comment:
I would allocate the exact amount of bytes for this instead of assigning from `innerPrefix`.

```go
hash := make([]byte, len(innerPrefix) + len(left) + len(right))
copy(...)
```